fms/modules/feedforward.py

```diff
@@ -118,11 +118,11 @@ def __init__(
         )
         self.setup_tp(rank, world_size)
 
-    def colwise_param_names(self) -> List[str]:
-        return ["w1"]
+    def colwise_params(self) -> Dict[str, List[int]]:
```

Comment:
are there any other cases where List[int] is going to make sense or is this still really a one-off for a single module type? if it's a one off, I still suspect there's a way we can refactor this more so that it's only seen in the actual place where it's needed.

If this truly needs to be a general purpose TP-feature, then I think it warrants more explanation/commenting. What does the 'int' even represent? I don't think it's truly "world_size" but some kind of shard size.

When someone later needs to implement another module type with a TP variant, they're going to need to clearly understand how to correctly override this, so we should keep it as simple and clear as possible.

Reply:
This is required any time we have weights that are fused when we want to tensor parallel. A list of size N for this implies there are N separate weights that could be stored together and may require a different form of tensor parallel. This is the most generic structure that could handle such cases. In the case of a non-fused weight, the list will be of size 1

Reply:
Latest commit contains docstring

Reply:
Also to clarify which modules this will help in the short term: Fused QKV in MHA, Fused MOE, Fused GatedLinearUnit